src/Service/Sqs/SqsConsumer.php

```diff
@@ -89,6 +90,9 @@ public function handleSqs(SqsEvent $event, Context $context): void
                     $stamps[] = new AmazonSqsXrayTraceHeaderStamp($context->getTraceId());
                 }
                 $this->busDriver->putEnvelopeOnBus($this->bus, $envelope->with(...$stamps), $this->transportName);
+            } catch (UnrecoverableExceptionInterface $exception) {
+                $this->logger->error(sprintf('SQS record with id "%s" failed to be processed. But failure was marked as unrecoverable. Message will be acknowledged.', $record->getMessageId()));
+                $this->logger->error($exception->getMessage());
```

Comment:
I would like to suggest to match the [messenger error level/message code](https://github.com/symfony/symfony/blob/6.2/src/Symfony/Component/Messenger/EventListener/SendFailedMessageForRetryListener.php#L82) a little:
```
$this->logger?->critical('Error thrown while handling message {class}. Removing from transport after {retryCount} retries. Error: "{error}"', $context + ['retryCount' => $retryCount, 'error' => $throwable->getMessage(), 'exception' => $throwable]);
```

```suggestion
                $this->logger->critical('Error thrown while handling message {class}. Removing from transport. Error: "{error}"', ['error' => $throwable->getMessage(), 'exception' => $exception]);
```

Reply:
I mostly be consistent with log in the other catch statement. But I'm OK to improve both of them ;)

Reply:
I see now I also made a mistake, the `{class}` is not yet set in context, please ignore my suggestion. I would like to see the exception in the `exception` [context](https://www.php-fig.org/psr/psr-3/#13-context)